godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/array/GodotArray.kt

```diff
@@ -0,0 +1,277 @@
+package godot.core
+
+import godot.gdnative.godot_array
+import kotlinx.cinterop.*
+import platform.posix.VISIT
+
+abstract class GodotArray<T> : NativeCoreType<godot_array>(), Iterable<T> {
```

Comment:
Maybe add a kDoc that this is also a class from which one should not derive?

Reply:
Or maybe one CAN derive from it to create custom godot array classes like one can in gdscript?
Not sure... what's your oppinion?

Reply:
It's fine to derive them. It won't break anything. Actually on another project using C++ bindings, I had to inherit from Array to do my own container so it helped me a lot. (I cheated by storing a C pointer as a Int inside a Godot Array because I didn't want to use the STL)

Reply:
:+1: